pkg/common/common.go

```diff
@@ -23,3 +23,43 @@ const (
 	// Resource lifecycle annotation value to prevent deletion.
 	OnRemoveKeep = "keep"
 )
+
+var Strategies = []DryRunStrategy{DryRunClient, DryRunServer}
+
+type DryRunStrategy int
+
+const (
```

Comment:
This duplicates the dry-run strategy in kubectl:

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go#L525

We probably don't want to duplicate this. But we can't use the kubectl one until we upgrade the version of kubectl we use to 1.18.x. We should talk about this.

Reply:
@seans3 By duplicating it now, we will easily be able to switch to kubectl DryRunStrategy when we use 1.18.x. We will sync up on this.

Reply:
We synced up offline. Our consensus sounds good. When we upgrade kubectl code to 1.18.x, we will merge the dry run strategy so there will only be one definition.